src/components/App/App.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {useState, useEffect} from "react";
+// @ts-ignore
+import logo from "./phish-logo.png";
+import { v4 as uuidv4 } from 'uuid';
+import "./App.css";
+
+// @ts-ignore
+import Search from "./Search/Search.jsx";
+import Song from "../../interfaces/Song";
+import GoogliResponse from "../../interfaces/GoogliResponse";
+
+const App = () => {
+  const [query, setQuery] = useState<string>('');
+  const [songs, setSongs] = useState<Song[]>([]);
+
+  const updateSearchQuery = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(e.target.value);
+    search();
+  };
+
+  const search = () => {
+    const filter = query;
+    const uuid = uuidv4();
+    setSongs([]);
+    fetch("https://googli-apparatus-backend.herokuapp.com/api/search/lyrics?uuid=" + uuid + "&filter=" + filter)
+      .then(
+        (response) => {
+          response.json().then((result: GoogliResponse) => {
+            setSongs(result.songs);
+          })
+        },
+        (error) => {
+          console.log(error);
+        }
+      );
+
+  };
+
+  useEffect(() => {
+    search();
+  }, [query]);
```

Comment:
This essentially says "any time the state value of `query` changes, run `search()`. 

I might need to dive into this deeper to fully mimic what you were had implemented here:

https://github.com/dolandrew/googli-ui/blob/ea703e0af88e071938b95cfa71c2cd30070fb5aa/src/components/App/App.jsx#L38-L40